src/lib/birthday/utils.ts

```diff
@@ -111,13 +111,11 @@ export function compareDate(month: Month, day: number, { now = Date.now() }: Tim
 export function getAge(data: DateWithOptionalYear, { now = Date.now() }: TimeOptions = {}) {
 	if (data.year === null) return null;
 
-	// If the birthday has happened, we subtract the years by one, that way:
-	//
-	// * 2021/03/18 - 2020/05/10 = 0
-	// * 2021/03/18 - 2020/02/26 = 1
-	const extra = compareDate(data.month, data.day, { now }) === -1 ? 0 : -1;
-	const years = new Date(now).getUTCFullYear() - data.year;
-	return years + extra;
+	const nowAsDate = new Date(now);
+	const dataAsDate = new Date(data.year, data.month - 1, data.day);
+
+	const years = new Date(Number(nowAsDate) - Number(dataAsDate)).getFullYear() - new Date(0).getFullYear() - 1;
```

Comment:
There is additionally no point using `new Date(now)` if you're simply going to convert it back into a number. `Date.now()` returns the same number you'd get from doing `Number(new Date(Date.now()))`, so there's no need to change `now` to a date.

Furthermore, that ternary seems to have been necessary - adding the unconditional -1 returns the wrong age for my birthday following these exact calculations, but I'm not quite sure the best way to fix that.
```suggestion
	const years = new Date(now - Number(dataAsDate)).getFullYear() - new Date(0).getFullYear() - 1;
```

Reply:
I changed it to getUTCFullYear and some issues raised while fixing some others, later i decided to use a "dumb" algorithm.

Reply:
I don't understand why when your initial solution needed a small tweak or two to be perfect.